db.py

```
@@ -60,32 +61,69 @@ def del_contact(self, id: int, contact: dict) -> tuple:
 
         pass
 
+
     def get_contacts(self, id: int) -> tuple:
         """return a list of contacts for the user"""
 
         pass
 
-    def get_user(self, id: int) -> tuple:
+    def get_user(self, username: str, password: str) -> tuple:
         """return user and error"""
+        
+        try:
+            self.db.execute('''SELECT * FROM user
+                WHERE username = ? AND password = ?''',
+                username, hash_password(password, username))
+
+            user = self.db.fetchone()
+
+            user = {
+                'id' : user[0],
+                'record_created' : user[1],
+                'last_logged_in' : user[2],
+                'username' : user[3],
+                'password' : user[4]
+            }
+
+            if not user: return (None, None, 401, 'Unauthorized')
+            else: return (user, auth.encode_token(user['id']), 0, None)
+        except sqlite3.Error as e:
+            return (None, None, 500, e)
 
-        pass
 
     def update_user_activity(self, id: int) -> tuple:
         """update user last_logged_in time"""
 
-        pass
+        time = str(datetime.utcnow())
+
+        try:
+            self.db.execute('''UPDATE user
+                SET last_logged_in = ? WHERE id = ?''', time, id)
+            self.con.commit()
+
+            self.db.execute('SELECT * FROM user WHERE id = ?', id)
+            return (self.db.fetchone(), 0, None)
```

We don't care about getting user info in this method, you can just return a error and message for both parts